### ISSUE TITLE
问题请教

### ISSUE THREAD

**Issue by yileicc** [zh320/realtime-semantic-segmentation-pytorch#23]
您好，非常感谢您能整理如此多的模型在一个文件中，然而，我想要训练我自己的数据集，我想要请教您一下，对于数据集的要求必须将mask.png文件转换成文件中的格式吗，我有多个类别但是ID号（0,1,2,4）不连续，且在训练过程中需要省略特殊的ID（4），应该怎么办呢

**Comment by zh320** [comment]
Hi,

Thank you for your interest in this repo. There’s no strict requirement for the label format. If you use a different format, you just need to implement your own dataset function, especially the `__getitem__` method. By the way, mask.png with class ids like 0,1,2,4 is already consistent with this repo, so no transformation is needed.

> 我有多个类别但是ID号（0,1,2,4）不连续，且在训练过程中需要省略特殊的ID（4），应该怎么办呢

You can train a 3-class model and set `ignore_index=4` in the cross-entropy loss.
